MetaMask/metamask-extension
# Snaps settings page

Closes: #11681 

Explanation:  Snaps settings page [[FIGMA](https://www.figma.com/file/nBKUQx0btfQiqVf30UA6KD/Flask-MVP?node-id=336%3A6342)] used to interact with the settings of individual snaps. There are two views: a list view under `SnapListTab` and a singular view with `ViewSnap`.  A user should be able to start & disable a snap and also be able to remove it entirely. They should also be able to disconnect a snap from a connected site.

**NOTE:** We do not have a snaps discover page / repository nor do we have snap verification in place (yet), so that part of the design was omitted for now. Some parts of the work were improvised due to the current design not covering all edge cases. Design will be polished in future iterations.

Manual testing steps:  
  - Start storybook
  - Navigate to `SettingsPage` and click on the "snaps" tab
  - Observe the changes
  - Additionally, it is recommended to observe the changes in a flask build: Run `yarn start --build-type flask` to start a dev build
